wiring/src/spark_wiring_cellular_printable.cpp

```diff
@@ -80,18 +77,49 @@ float CellularSignal::getQualityValue() const
     return 0.0f;
 }
 
-// TODO: remove once rssi/qual are removed
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 size_t CellularSignal::printTo(Print& p) const
 {
     size_t n = 0;
-    n += p.print((*this).rssi, DEC);
+    n += p.print(this->getStrengthValue(), 2);
     n += p.print(',');
-    n += p.print((*this).qual, DEC);
+    n += p.print(this->getQualityValue(), 2);
     return n;
 }
-#pragma GCC diagnostic pop
+
+bool CellularSignal::isAccessTechnologyValid() const
+{
+    return (sig_.rat != NET_ACCESS_TECHNOLOGY_NONE);
+}
+
+bool CellularSignal::isSignalStrengthValid() const
+{
+    return (sig_.rssi != std::numeric_limits<int32_t>::min());
+}
+
+bool CellularSignal::isSignalQualityValid() const
+{
+    return (sig_.qual != std::numeric_limits<int32_t>::min());
+}
+
+bool CellularSignal::isSignalQualitySupported() const
+{
+#if (PLATFORM_ID == PLATFORM_ELECTRON_PRODUCTION)
+    return (sig_.rat == NET_ACCESS_TECHNOLOGY_GSM) ? false : true;
```

Comment:
How accurate is this? The cellular HAL seems to report BER as the signal quality in case of GSM.